ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/viewmodel/WalletViewModel.kt

```diff
@@ -371,9 +376,14 @@ private fun Synchronizer.toTransactions() =
         receivedTransactions.distinctUntilChanged()
     ) { cleared, pending, sent, received ->
         // TODO [#157]: Sort the transactions to show the most recent
+        // TODO [#157]: https://github.com/zcash/secant-android-wallet/issues/157
+
+        Twig.debug { "Test: $cleared, $pending" }
+        // Fixme: How to combine these lists of unrelated model classes?
+        // Fixme: TransactionOverview x PendingTransaction x Transaction
         buildList<Transaction> {
-            addAll(cleared)
-            addAll(pending)
+            // addAll(cleared)
+            // addAll(pending)
```

Comment:
What should be the right solution here? I can image some connection between transaction model classes (inheritance x common interface). Or we can add mapping functions between these model classes, as they have many common attributes.

Reply:
Yes this is an area where the SDK needs work but we probably won't resolve it with this release.

We can probably stop querying for sent/received transactions, as cleared transactions is inclusive of sent/received.

For now, I think this could return either a list of TransactionOverview objects or a custom object defined by the app (until we update the SDK).  Given that sent/received can be eliminated, then you're just dealing with TransactionOverview versus PendingTransaction.

Reply:
And in the case of working with the `TransactionOverview` objects here. May the SDK provide us a mapping function between `PendingTransaction` and `TransactionOverview`? Something like:
```
fun PendingTransaction.toTransactionOverview(): TransactionOverview {
    return TransactionOverview(
        id = id,
        raw = raw,
        ...
    )
}
```

Reply:
I think this would lead to runtime breakage in clients.

`TransactionOverview` is used to query for things in the Synchronizer and a converted PendingTransaction wouldn't be valid for that usage since the id would belong to a different database.

Until we have time to revise PendingTransaction it seems like this might need to be dealt with on the client side. Thoughts?

Reply:
Understand. I just pushed a commit, which brings a common transaction model class `CommonTransaction`, which is a wrapper for all attributes from both `PendingTransaction` and `TransactionOverview`. So we can now decide what we should keep and what can be omitted for our use cases.